design/seed_vm.md

```diff
@@ -0,0 +1,190 @@
+| ---------------- | ------------------------------------------------ |
+| PC               | Program Counter                                  |
+| R(A), R(B), R(C) | Register indexed by A, B or C                    |
+| RK(B), RK(C)     | Register or constant indexed by B or C           |
+| Kst(n)           | Constant in the constant list indexed by n       |
+| Gbl[sym]         | Global variable named with the string symbol sym |
+| sBx              | Signed offset for JMP instruction                |
+
+For RK(B) and RK(C), the MSB (most significant bit) of B or C is used to
+indicate if B or C is the index of the register or constant:
+
+- MSB == 0: B or C is the index of the register
+- MSB == 1: (B - 256) or (C - 256) is the index of the constant in the constant
+  list
+
+## Instruction Set
+
+All the SeedLang instructions are listed as follows:
+
+| Opcode | Name     | Description                                         |
+| :----: | -------- | --------------------------------------------------- |
+|   0    | MOVE     | Copy a value between registers                      |
+|   1    | LOADBOOL | Load a boolean value into a register                |
+|   2    | LOADK    | Load a constant into a register                     |
+|   3    | GETGLOB  | Read a global variable into a register              |
+|   4    | SETGLOB  | Write a register value into a global variable       |
+|   5    | GETELEM  | Read a list or table element into a register        |
+|   6    | SETELEM  | Write a register value into a list or table element |
+|   7    | ADD      | Addition operation                                  |
+|   8    | SUB      | Subtract operation                                  |
+|   9    | MUL      | Multiply operation                                  |
+|   10   | DIV      | Divide operation                                    |
+|   11   | MOD      | Modulus (reminder) operation                        |
+|   12   | POW      | Exponentiation operation                            |
+|   13   | UNM      | Unary minus operation                               |
+|   14   | NOT      | Logical not operation                               |
+|   15   | LEN      | Length operation                                    |
+|   16   | JMP      | Unconditional jump                                  |
+|   17   | EQ       | Equality test                                       |
+|   18   | LT       | Less than test                                      |
+|   19   | LE       | Less than or equal to test                          |
+|   20   | TEST     | Boolean test, with conditional jump                 |
+|   21   | TESTSET  | Boolean test, with conditional jump and assignment  |
+|   22   | CALL     | Call a function                                     |
+|   23   | RETURN   | Return from a function call                         |
+
+### Move and Load Constant
+
+```shell
+MOVE A B                    // R(A) := R(B)
```

Comment:
Since you have annotated this markdown code block as a "shell" code, it's better to start the comments inside the code with a real shell programming notation, i.e., "#", instead of "//". By using "#", the rendered markdown doc will look pretty when the comments are highlighted.

Same for the following instructions.